geotrek/signage/templates/signage/sql/post_10_views.sql

```diff
@@ -86,37 +86,21 @@ LEFT JOIN ( WITH signage_condition AS (
 LEFT JOIN signage_sealing d ON a.sealing_id = d.id
 LEFT JOIN authent_structure e ON a.structure_id = e.id
 LEFT JOIN common_accessmean i ON a.access_id = i.id
-LEFT JOIN
-    (SELECT array_to_string(ARRAY_AGG (b.name ORDER BY b.name), ', ', '_') zoning_city,
-            a.id
-     FROM
-         (SELECT e.id,
-                 e.geom
-          FROM signage_signage t,
-               signage_signagetype b,
-               core_topology e
-          WHERE t.topo_object_id = e.id
-              AND t.type_id = b.id
-              AND e.deleted = FALSE ) a
-     JOIN zoning_city b ON ST_INTERSECTS (a.geom, b.geom)
-     GROUP BY a.id) f ON a.id = f.id
-LEFT JOIN
-    (SELECT array_to_string(ARRAY_AGG (b.name ORDER BY b.name), ', ', '_') zoning_district,
-            a.id
-     FROM
-         (SELECT e.id,
-                 e.geom
-          FROM signage_signage t,
-               signage_signagetype b,
-               core_topology e
-          WHERE t.topo_object_id = e.id
-              AND t.type_id = b.id
-              AND e.deleted = FALSE ) a
-     JOIN zoning_district b ON ST_INTERSECTS (a.geom, b.geom)
-     GROUP BY a.id) g ON a.id = g.id
+LEFT JOIN LATERAL (
+     SELECT array_to_string(array_agg(b_1.name ORDER BY b_1.name), ', '::text, '_'::text) AS zoning_city
+           FROM   zoning_city b_1
+            WHERE st_intersects(a.geom, b_1.geom)
+          GROUP BY a.id
+    ) f ON true
```

Comment:
**Suggestion:** Consider using `ST_Within` instead of `ST_Intersects` for spatial intersection checks. `ST_Within` can provide better performance as it doesn't need to calculate the actual intersection geometry. [performance]
```suggestion
LEFT JOIN LATERAL (
     SELECT array_to_string(array_agg(b_1.name ORDER BY b_1.name), ', '::text, '_'::text) AS zoning_city
           FROM   zoning_city b_1
            WHERE st_within(a.geom, b_1.geom)
          GROUP BY a.id
    ) f ON true
```